Failed compiling the bootloader on old Linux 4.4 & 4.5

I'm building Linux apps using an old `Ubuntu 10` docker to support old GLIBC versions.
The latest 4.4 and 4.5 releases are not working for me failing to build the bootloader:
```
  'configure' finished successfully (0.550s)
  'make_all' finished successfully (0.003s)
  Waf: Entering directory `/tmp/pip-install-xni3ihz7/pyinstaller_22eb48a10e1f4419b7b04637fbe24337/bootloader/build/debug'
  [ 1/15] Compiling src/pyi_python.c
  [ 2/15] Compiling src/pyi_global.c
  [ 3/15] Compiling src/pyi_archive.c
  [ 4/15] Compiling src/pyi_win32_utils.c
  [ 5/15] Compiling src/pyi_path.c
  [ 6/15] Compiling src/pyi_splashlib.c
  ../../src/pyi_archive.c: In function '_pyi_find_cookie_offset':
  ../../src/pyi_archive.c:417: error: 'for' loop initial declarations are only allowed in C99 mode
  ../../src/pyi_archive.c:417: note: use option -std=c99 or -std=gnu99 to compile your code

  Waf: Leaving directory `/tmp/pip-install-xni3ihz7/pyinstaller_22eb48a10e1f4419b7b04637fbe24337/bootloader/build/debug'
  Build failed
   -> task in 'OBJECTS' failed with exit status 1 (run with -v to display more information)
  No precompiled bootloader found. Trying to compile it for you ...
  ERROR: Failed compiling the bootloader. Please compile manually and rerun setup.py
  ----------------------------------------
  ERROR: Failed building wheel for pyinstaller
Failed to build pyinstaller
ERROR: Could not build wheels for pyinstaller which use PEP 517 and cannot be installed directly
```

I tried compiling it manually but it also fails, how can I get it to work? (I have reverted to 4.3 for now that still works)

PyInstaller: 4.4/4.5
Python: 3.7.9
OS: Linux

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -59,6 +59,14 @@ jobs:
           path: ${{ env.pip_cache_dir }}
           key: ${{ runner.os }}-${{ matrix.python-version }}
 
+      - name: Check if bootloader code conforms to gnu90 C standard
+        if: startsWith(matrix.os, 'ubuntu')
+        run: |
+          # Compile bootloader
+          cd bootloader
+          CC="gcc -std=gnu90" python waf all
+          cd ..
+
       - name: Set up environment
         run: |
           # Update pip.

--- a/bootloader/src/pyi_archive.c
+++ b/bootloader/src/pyi_archive.c
@@ -394,7 +394,7 @@ _pyi_find_cookie_offset(FILE *fp)
     /* Search the file back to front, in overlapping SEARCH_CHUNK_SIZE
      * chunks. */
     do {
-        size_t chunk_size;
+        size_t chunk_size, i;
         start_pos = (end_pos >= SEARCH_CHUNK_SIZE) ? (end_pos - SEARCH_CHUNK_SIZE) : 0;
         chunk_size = (size_t)(end_pos - start_pos);
 
@@ -414,7 +414,7 @@ _pyi_find_cookie_offset(FILE *fp)
         }
 
         /* Scan the chunk */
-        for (size_t i = chunk_size - MAGIC_SIZE + 1; i > 0; i--) {
+        for (i = chunk_size - MAGIC_SIZE + 1; i > 0; i--) {
             if (_pyi_match_magic(buffer + i - 1)) {
                 offset = start_pos + i - 1;
                 goto cleanup;

